src/include/mempool.h

```diff
@@ -255,11 +255,16 @@ class pool_t {
 
   void adjust_count(ssize_t items, ssize_t bytes);
 
-  shard_t* pick_a_shard() {
+  static size_t pick_a_shard_int() {
     // Dirt cheap, see:
-    //   http://fossies.org/dox/glibc-2.24/pthread__self_8c_source.html
+    //   https://fossies.org/dox/glibc-2.32/pthread__self_8c_source.html
     size_t me = (size_t)pthread_self();
-    size_t i = (me >> 3) & ((1 << num_shard_bits) - 1);
+    size_t i = (me >> 12) & ((1 << num_shard_bits) - 1);
```

Comment:
main()
{
  printf("%p %p\n", __builtin_thread_pointer(), pthread_self());
}

with 4K pagesize：
for i in {0..10} ; do ./test ; done
0xffff8f0ea6f0 0xffff8f0ea000
0xffffb63ac6f0 0xffffb63ac000
0xffff9a0cc6f0 0xffff9a0cc000
0xffff81cb26f0 0xffff81cb2000
0xffffb80c56f0 0xffffb80c5000
0xffffb94de6f0 0xffffb94de000
0xffffbee2b6f0 0xffffbee2b000
0xffffa21d86f0 0xffffa21d8000
0xffff9dc306f0 0xffff9dc30000
0xffffb0be26f0 0xffffb0be2000
0xffffa2ec36f0 0xffffa2ec3000

with 64K pagesize
0xfffc1a9930c0 0xfffc1a9929c0
0xfffe73bf30c0 0xfffe73bf29c0
0xfffbd05330c0 0xfffbd05329c0
0xfffd8a6a30c0 0xfffd8a6a29c0
0xfffd2a2b30c0 0xfffd2a2b29c0
0xffff89a930c0 0xffff89a929c0
0xfffedb5b30c0 0xfffedb5b29c0
0xfffd504630c0 0xfffd504629c0
0xfffe15ca30c0 0xfffe15ca29c0
0xfffc09ac30c0 0xfffc09ac29c0
0xfffed68b30c0 0xfffed68b29c0

The hash algorithm can be correctly hashed only when different page sizes are shifted based on the page offset.
4K with 12,64K with 16